homeassistant/components/zha/__init__.py

```diff
@@ -369,6 +410,104 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZHAEvent(object):
+    """When you want signals instead of entities.
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Register callback that will be used for signals."""
+        self._hass = hass
+        self._cluster = cluster
+        self._cluster.add_listener(self)
+        self._ieee = discovery_info['endpoint'].device.ieee
+        self._ieeetail = ''.join(['%02x' % (o, ) for o in self._ieee[-4:]])
+        if discovery_info['manufacturer'] and discovery_info['model'] is not \
+                None:
+            self._id = "{}.{}_{}_{}_{}{}".format(
+                'zha',
+                slugify(discovery_info['manufacturer']),
+                slugify(discovery_info['model']),
+                self._ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+        else:
+            self._id = "{}.zha_{}_{}{}".format(
+                'zha',
+                self._ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+
+
+class ZHASwitchEvent(ZHAEvent):
+    """Switch / remote event for zha"""
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Initialize Switch."""
+        super().__init__(hass, cluster, discovery_info)
+
+    @callback
+    def cluster_command(self, tsn, command_id, args):
+        """Handle commands received to this cluster."""
+        if command_id in (0x0000, 0x0040):
+            self._hass.bus.fire('zha.off', {'device': self._id},
```

Comment:
Please extract the event names as `EVENT_XXX` constants at the top

Reply:
Will do